pkg/common/auto_detect.go

```diff
@@ -1,8 +1,6 @@
 package common
 
 import (
-	"k8s.io/apimachinery/pkg/runtime/schema"
```

Comment:
Just out of curiosity, why are we removing this package across the project?

Reply:
I simplified the auto detecting mechanism. It turned out it contained a race condition (between the reconciler and the channel in the `auto_detect.go`). Once I did that, the schema started to be useless. 

I plan to simplify it a bit more in the future. I'd like to squash Auto Detect mechanism with the Background object we have. Currently, it's hard to wrap your head around it...